scripts/topic_repos.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import secrets
+import subprocess
+import sys
+from pathlib import Path
+
+import yaml
+from github import Github
+from pydantic import BaseModel, SecretStr
+from pydantic_settings import BaseSettings
+
+
+class Settings(BaseSettings):
+    github_repository: str
+    github_token: SecretStr
+
+
+class Repo(BaseModel):
+    name: str
+    html_url: str
+    stars: int
+    owner_login: str
+    owner_html_url: str
+
+
+def main() -> None:
+    logging.basicConfig(level=logging.INFO)
+    settings = Settings()
+
+    logging.info(f"Using config: {settings.model_dump_json()}")
+    g = Github(settings.github_token.get_secret_value(), per_page=100)
+    r = g.get_repo(settings.github_repository)
+    repos = g.search_repositories(query="topic:fastapi")
+    repos_list = list(repos)
+    final_repos: list[Repo] = []
+    for repo in repos_list[:100]:
+        if repo.full_name == settings.github_repository:
+            continue
+        final_repos.append(
+            Repo(
+                name=repo.name,
+                html_url=repo.html_url,
+                stars=repo.stargazers_count,
+                owner_login=repo.owner.login,
+                owner_html_url=repo.owner.html_url,
+            )
+        )
+    data = [repo.model_dump() for repo in final_repos]
+    logging.info("Finished")
+
+    # Local development
+    # repos_path = Path("../docs/en/data/topic_repos.yml")
+    repos_path = Path("./docs/en/data/topic_repos.yml")
+    repos_old_content = repos_path.read_text(encoding="utf-8")
+    new_repos_content = yaml.dump(data, sort_keys=False, width=200, allow_unicode=True)
+    if repos_old_content == new_repos_content:
+        logging.info("The data hasn't changed. Finishing.")
+        sys.exit(0)
```

Comment:
```suggestion
        return
```

I see this was from a copy of the other scripts that were run with `if __name__ == "__main__"`, but as this is now in a function, we can just return to finish the execution here, we don't have to call `sys.exit(0)`